Removed all refrences to access wideners (#468)
minecraft-access/minecraft-access: CONTRIBUTING.md

Java programming, concepts of writing maintainable elegant code,
git operations, basic knowledge of Linux shell commands, collaboration with GitHub...
If you are not familiar with how to clone a project and submit a Pull Request using GitHub,
please

> read [the GitHub official documentation](https://docs.github.com/en/get-started/quickstart/contributing-to-projects).

Reviewer: there is a next line behind "please"

Then there is some knowledge for Minecraft modding you'd like to learn before coding.
You're better off learning by doing, first git clone the project and set up the local environment,
then read the project files, start the mod in debug mode and set breakpoints to understand the order of execution.

This project uses [Gradle](https://docs.gradle.org/current/userguide/getting_started_eng.html) as dependency manager and

> Groovy DSL

Reviewer: shouldn't has a next line ahead of "Groovy DSL"

with the help of [Architectury Loom](https://docs.architectury.dev/loom/introduction),
a framework for multi-platform modding.
Derivatives from [Fabric Loom](https://fabricmc.net/wiki/documentation:fabric_loom).
Architectury Loom functions as a Gradle plugin and mod configurator, it won't affect the code.
For questions about architectury,
you can ask in [their Discord server](https://discord.architectury.dev/) instead of searching for documentation,
but it doesn't harm to search before asking.
Since Architectury Loom is a derivative of Fabric Loom,
the Fabric Loom's documentation also applies to Architectury Loom in most cases,
which is more detail than Architectury Loom's.

Then the core framework we heavily depend on - [SpongePowered Mixin](https://github.com/SpongePowered/Mixin/wiki),
`Mixin` in short, a framework for modifying Java bytecode.
For the convenience of reusing the same set of code for all platforms,
this mod doesn't directly depend on Forge API or Fabric API;
instead we directly modify original game code,
injecting our logic into it.
We also depend on the Architectury API to handle functions that are different for each platform with one piece of code for both.

> Mixins are the same principle as a computer virus or cheating tools, you add new code at the head or tail (start or end) of the code that is already being ran by the program.

Reviewer: > you add new code at the head or tail (start or end)

Nope, there are more positions, so just say "you inject new logic into original program"

Well, to be precise, 95% of the code is platform-independent.
Mixin is a complex but handy framework;
for learning it, I recommend
reading [the tutorial by Fabric](https://fabricmc.net/wiki/tutorial:mixin_introduction)

> and [the official wiki](https://github.com/SpongePowered/Mixin/wiki).

Reviewer: ```suggestion
, [`Mixin`'s official wiki](https://github.com/SpongePowered/Mixin/wiki) and [wiki of `MixinExtras` library](https://github.com/LlamaLad7/MixinExtras/wiki).
```

Please note that the official wiki is very technical and detailed and is not suitable for usage as a user manual,
use it as an API reference.
For questions about Mixin,

> you can ask in `mod-dev` channels of [the Fabric Discord server](https://discord.com/invite/v6v4pMv) or the `mixin`

Reviewer: unwanted new line
```suggestion
you can ask in `mod-dev` channels of [the Fabric Discord server](https://discord.com/invite/v6v4pMv) or the `mixin` channel of the [NeoForge Discord server](https://discord.com/invite/neoforged),
```

> channel of the [NeoForge Discord server](https://discord.com/invite/neoforged),

Reviewer: ```suggestion
```

they are a friendly bunch and willing to answer any questions you may have.

Finally, to search for a suitable injection point,
we need to [read the original code](https://fabricmc.net/wiki/tutorial:reading_mc_code).

> Thanks to the [MojMap mappings](https://minecraft.wiki/w/Obfuscation_map) tool we can read the source code

Reviewer: ```suggestion
Thanks to the [MojMap mappings](https://minecraft.wiki/w/Obfuscation_map) tool we can read the source code that de-obfuscated from the game's bytecode.
```

> that de-obfuscated from the game's bytecode.

Reviewer: ```suggestion
```

You don't need to manually operate the mappings, Loom will do it for you while running related gradle tasks.

> If you'd like

Reviewer: ```suggestion
If you'd like to know some terms of Minecraft project structure like [`client side` vs `server side`](https://fabricmc.net/wiki/tutorial:side), this mod is a pure client side mod for now.
```

> to know some terms of Minecraft project structure like [`client side` vs

Reviewer: ```suggestion
```

> `server side`](https://fabricmc.net/wiki/tutorial:side),

Reviewer: ```suggestion
```

> this mod is a pure client side mod for now.

Reviewer: ```suggestion
```

## Setting up Local Environment

Java 21 is required to be pre-installed for this project.
IntelliJ IDEA is recommended since it has some convenient Minecraft modding plugins:

> [Architectury](https://plugins.jetbrains.com/plugin/16210-architectury)

Reviewer: ```suggestion
[Architectury](https://plugins.jetbrains.com/plugin/16210-architectury) and [Minecraft Development](https://plugins.jetbrains.com/plugin/8327-minecraft-development).
```

> and [Minecraft Development](https://plugins.jetbrains.com/plugin/8327-minecraft-development).

Reviewer: ```suggestion
```

Can't find corresponding vscode extensions, but since Gradle is platform-independent, you can use any IDE that you want.

> Now clone the project and run `gradlew build`,

Reviewer: ```suggestion
Now clone the project and run `gradlew build`, to let Gradle automatically download dependencies, it takes a few minutes to set up the first time.
```

> to let Gradle automatically download dependencies, it takes a few minutes to set up the first time.

Reviewer: ```suggestion
```

> After finishing the first build, you should run `gradlew genSources` to generate the de-obfuscated source code,

Reviewer: ```suggestion
After finishing the first build, you should run `gradlew genSources` to generate the de-obfuscated source code, set generated source jar as compiled jar's source according to the `Generating Minecraft Sources` section of the [tutorial](https://fabricmc.net/wiki/tutorial:setup).
```

> set generated source jar as compiled jar's source according to the `Generating Minecraft Sources` section of

Reviewer: ```suggestion
```

> the [tutorial](https://fabricmc.net/wiki/tutorial:setup).

Reviewer: ```suggestion
```

Now you can explore the project.

> For IntelliJ, Loom will generate

Reviewer: ```suggestion
For IntelliJ, Architectury Loom will generate some [Run Configurations](https://www.jetbrains.com/help/idea/run-debug-configuration.html) like `Minecraft Client (:fabric)`, `Minecraft Server (:fabric)`,  `Minecraft Client (:neoforge)`, `Minecraft Server (:neoforge)`, you can tell the purpose of each configuration by their name, for example, `Minecraft Client (:fabric)` is for starting the client side game on Fabric.
```

> some [Run Configurations](https://www.jetbrains.com/help/idea/run-debug-configuration.html) like

Reviewer: ```suggestion
```

> `Minecraft Client (:fabric)`, `Minecraft Server (:fabric)`,  `Minecraft Client (:neoforge)`,

Reviewer: ```suggestion
```

> `Minecraft Server (:neoforge)`, you can tell the purpose of each configuration by their name, for example,

Reviewer: ```suggestion
```

> `Minecraft Client (:fabric)` is for starting the client side game on Fabric.

Reviewer: ```suggestion
```

> PS: If the Run Configurations get broken accidentally,

Reviewer: ```suggestion
PS: If the Run Configurations get broken accidentally, delete `.idea/runConfigurations` then re-open the project in IntelliJ. Loom will re-generate them automatically.
```

> delete `.idea/runConfigurations` then re-open the project in IntelliJ. Loom will re-generate them automatically.

Reviewer: ```suggestion
```

### File Structure

* `.github`: GitHub related stuff like workflow (CI) files.
* `gradle`: Mainly contains Gradle wrapper configuration.
* `docs`: Project documentations.

> * `build.gradle`: Root Gradle build script of this Gradle managing project. ([Gradle doc about

Reviewer: ```suggestion
* `build.gradle`: Root Gradle build script of this Gradle managing project. ([Gradle doc about `build.gradle` file](https://docs.gradle.org/current/userguide/build_file_basics.html))
```

>   `build.gradle` file](https://docs.gradle.org/current/userguide/build_file_basics.html))

Reviewer: ```suggestion
```

* `gradle.properties`: Configuration about mod and dependency versioning.

> * `settings.gradle`: Defines that this project has three subprojects: `common`, `fabric`, `neoforge`. ([Gradle doc about

Reviewer: ```suggestion
* `settings.gradle`: Defines that this project has three subprojects: `common`, `fabric`, `neoforge`. ([Gradle doc about `settings.gradle` file](https://docs.gradle.org/current/userguide/settings_file_basics.html))
```

>   `settings.gradle` file](https://docs.gradle.org/current/userguide/settings_file_basics.html))

Reviewer: ```suggestion
```

* `common`: Gradle subproject, Platform-independent code and configurations, normal Java project structure.

>     * `build.gradle`: Gradle build script of `common` subproject, it has developing dependencies like `lombok`, `log4j`,

Reviewer: ```suggestion
    * `build.gradle`: Gradle build script of `common` subproject, it has developing dependencies like `lombok`, `log4j`, `junit`.
```

>       `junit`.

Reviewer: ```suggestion
```

`Functional Core, Imperative Shell`](https://news.ycombinator.com/item?id=34860164),
[`Red - Green - Refactor`](https://martinfowler.com/bliki/TestDrivenDevelopment.html)
and more described in [this article](https://www.amazingcto.com/mocking-is-an-antipattern-how-to-test-without-mocking/).

## CI

Since this project is hosted on GitHub, it's natural for us to choose GitHub Actions as the CI system.
This project has an automatic test-build-release pipeline.

* For every PR and build, [`test` workflow](.github/workflows/test.yml) will be triggered for running the test suite
  against the current code.
* The [`build` workflow](.github/workflows/build.yml) is responsible for building all platforms mod jars, you can
  manually dispatch it in the GitHub Actions Page of the repo.
* When a new version is ready and merged, we'll manually run the [`release` workflow](.github/workflows/release.yml) to
  create a new pre-release in GitHub Releases, after one-week exposure, we'll run this workflow against this version
  again to change the GitHub release to a formal state, and publish the mod to Modrinth and Curse Forge.

## Developing Tips

> * You should enable this mod's `Other Configurations - Debug Mode` configuration to let the mod print debug level

Reviewer: ```suggestion
* You should enable this mod's `Other Configurations - Debug Mode` configuration to let the mod print debug level logs.
```

>   logs. This may automatically trigger if running in a development environment based on Architectury API's detection.

Reviewer: ```suggestion
This may automatically trigger if running in a development environment based on Architectury API's detection.
```

> * When running the game with your IDE, the main directory is `{platform directory}/run`. This has the same structure as in the production

Reviewer: ```suggestion
* When running the game with your IDE, the main directory is `{platform directory}/run`. This has the same structure as in the production environment).
```

>   environment). You can see familiar directories like `config`, `mods`, `saves` under it.

Reviewer: ```suggestion
You can see familiar directories like `config`, `mods`, `saves` under it.
```

> * You can run `gradlew {platform}:build` to build the mod jar for a specific platform (when root Gradle `build` task fails for

Reviewer: ```suggestion
* You can run `gradlew {platform}:build` to build the mod jar for a specific platform (when root Gradle `build` task fails for some reason), built mod jars are under `{platform directory}/build/libs`, the one without a suffix.
```

>   some reason), built mod jars are under `{platform directory}/build/libs`, the one without a suffix.

Reviewer: ```suggestion
```

When you decide to make contributions, we expect you to comment in the corresponding issue,
make a new issue, or talk in the `mc-dev` channel of our [Discord server](https://discord.gg/yQjjsDqWQX).
Industry experience tells us that shift-left checks and discussions are good for software development —
for example, we can provide more relevant knowledge, project details, and advice to help you do things better.
It can only be done if we know what you want to do, though.

Of course there is no mandatory requirement
(praise [the freewheeling open source community](http://www.catb.org/~esr/writings/cathedral-bazaar/)!),
the code for this project is open source under an opensource license.
This means that you do not need to get anyone's permission to make changes to your fork of this project,
you just need to make it work for **your personal needs**.
If you feel that your changes will also benefit the upstream (main project), you can submit a Pull Request to [the main repository](https://github.com/minecraft-access/minecraft-access),
and only then does the `contributing` begin.

[Weblate]: https://hosted.weblate.org/engage/minecraft-access/

> 

Reviewer: ```suggestion
```
